xtask/src/main.rs

```diff
@@ -3,6 +3,8 @@
 //! This binary is integrated into the `cargo` command line by using an alias in
 //! `.cargo/config`. Run commands as `cargo xtask [command]`.
 
+#![allow(clippy::exhaustive_structs)]
```

Comment:
Was this triggering on `Metadata` below? I think we can just remove `pub` from that.

Reply:
Yeah, I had to make it `pub(crate)` so the method that uses it didn't complain.